build-tools/packages/build-cli/src/lib/bump.ts

```diff
@@ -15,6 +15,35 @@ import {
 } from "@fluid-tools/version-tools";
 import * as semver from "semver";
 
+/**
+ * A type representing the types of dependency updates that can be done. This type is intended to match the type
+ * npm-check-updates uses for its `target` argument.
+ */
+export type DependencyUpdateType =
+    | "latest"
+    | "newest"
+    | "greatest"
+    | "minor"
+    | "patch"
+    | `@${string}`;
+
+/**
+ * A type guard used to determine if a string is a DependencyUpdateType.
+ *
+ * @internal
+ */
+export function isDependencyUpdateType(str: string | undefined): str is DependencyUpdateType {
+    if (str === undefined) {
+        return false;
+    }
+
+    if (["latest", "newest", "greatest", "minor", "patch"].includes(str)) {
```

Comment:
Nit: Should we generate a static array of flags from the enum itself? Otherwise we'll have to update this any time we update the enum. I guess we would have to split the current type into 2 to do this reasonably though (1 for the pre-defined kinds, and 1 that includes the dynamic `@foo` kind). So feel free to disregard this :)

Reply:
Yeah, if only union types were iterable (though how would you iterate the `@foo` ones?)! I was hoping to reuse a type from npm-check-updates but they don't export it, and it includes a function signature.